fast_response/FastResponseAnalysis.py

```diff
@@ -992,6 +1020,424 @@ def unblind_TS(self, custom_events=None):
+    def fluxToNs_decDependent(self,flux,dec):
+        mean_signal=self.fluxToAverageNs(flux)
+        acceptance_weights = self.inj.relative_signal_acceptance(np.sin(dec))
+        num=[np.array(np.around(w * mean_signal / float(self.prior.nprior)),
+                            dtype=np.int).tolist()
+                   for w in acceptance_weights]
+        num_sum = np.sum(num, dtype=int)
+        return num_sum
+    
+    def make_posterior_map(self, fluxToTest=[1,2,3,4,5], 
+                           prior_func=lambda ns,gamma,ra,dec: 1, 
+                           plotting_location=None,
+                           custom_events=None,
+                           useFlux=False
+                           ):
+        # Posterior Approach
+        # We want a function P(RA, DEC | X) where X is the neutrino data
+        # Compute the entire likelihood space P(X | RA, DEC, ns, gamma)
+        # Apply Bayes rule, marginalize over nuisance parameters to get P(RA, DEC | X)
+ 
+        #flux/ns to test: given as list  
+        #Give prior as a function f(ns, gamma, ra, dec)
+            #As currently implemented,the llh will have gamma must be constant set by the self.index
+            #Prior flat in ns and spatially by default. 
+
+        #plotting_location is the directory where plots should be output
+        #custom events will be injected
+        #useFlux determines whether fluxToTest is treated as ns or flux
+
+
+        t1 = time.time()
+        val=None
+        skymap = self.skymap
+        
+        
+        spatial_prior = SpatialPrior(
+            skymap, 
+            containment = self._containment,allow_neg=self._allow_neg)
+        
+        nside=self.nside
+        decs_list = hp.pix2ang(nside, np.arange(hp.nside2npix(nside)))
+
+
+        test_flux=1e-17
+        if(useFlux):
+            mean_flux_arr=fluxToTest
+            nsPerFluxArr=[self.fluxToNs_decDependent(test_flux,d)/test_flux for d in decs_list]
+        else:
+            #work in uniform ns
+            mean_flux_arr=fluxToTest
+            nsPerFluxArr=[1]*12*pow(nside,2) 
+
+
+        for fluxT in mean_flux_arr:
+            temp_val = self.llh.scan(0.0,0.0, scramble = False,spatial_prior=spatial_prior,
+                    time_mask = [self.duration/2., self.centertime],
+                    pixel_scan=[nside, self._pixel_scan_nsigma],
+                    fixed=['nsignal'],
+                    nsignal=np.array([fluxT*conversion for conversion in nsPerFluxArr],dtype=float), 
+
+                    inject=custom_events
+                    )
+            if val is None:
+                val=temp_val
+            else:
+                val=np.hstack((val,temp_val))
+        ras=val["ra"]
+        decs=val["dec"]
+        nss=val["nsignal"]
+    
+        #Placeholder: not required in fixed index analyses, but future improvements could expand this
+        if "gamma" in val.dtype.names:
+            gammas=val["gamma"]
```

Comment:
we are not currently using unfixed gamma in the posterior, correct? since you are not marginalizing over it, I think you can require a fixed gamma (using an `assert` at the beginning of this function)